### ISSUE TITLE
False positives with SCSS Nested Properties in property-no-unknown

### ISSUE THREAD

**Issue by epmatsw** [stylelint/stylelint#1870]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where valid CSS is being reported as an issue.

> Which rule, if any, is this issue related to?

`property-no-unknown`

> What CSS is needed to reproduce this issue?

e.g.

```css
.test {
    border: {
            style: solid;
            color: red;
    }
}
```

> What stylelint configuration is needed to reproduce this issue?

stylelint-config-standard 13.0.0

> Which version of stylelint are you using?

7.2.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Scripted via npm (`stylelint -q -s 'scss' "src/**/*.css"`)

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, SCSS nested properties.

> What did you expect to happen?

No errors raised.

> What actually happened (e.g. what warnings or errors you are getting)?

The following errors were flagged:

```shell
Unexpected unknown property "style"   property-no-unknown
```

**Comment by evilebottnawi** [comment]
I think we should ignore this construction, `stylelint-scss` can write rule for check this.

**Comment by jeddy3** [comment]
@epmatsw Thanks for report and for using the template.

@evilebottnawi Yes. Unlike nested selectors, I don't think we should resolve nested properties in core as it's a non-standard language extension.

As always, lets work out a way to ignore this construct so this rule doesn't trip up on it.

Shall we add a conditional to [`isStandardSyntaxProperty`](https://github.com/stylelint/stylelint/blob/master/src/utils/isStandardSyntaxProperty.js)? 

I think returning early if the *selector* of the *parent rule* ends with `:` and doesn't start with `--` (to exclude custom property sets). Can you any foreseen any problems with this heuristic?

**Comment by evilebottnawi** [comment]
@jeddy3 sgtm

**Comment by evilebottnawi** [comment]
@stylelint/core any objections?

**Comment by davidtheclark** [comment]
Sounds great.

**Comment by jeddy3** [comment]
@epmatsw Would you like to [contribute this](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs) fix?

**Comment by epmatsw** [comment]
@jeddy3 yeah, I can take a stab at it this afternoon

**Comment by epmatsw** [comment]
@jeddy3 Yeah, I'm not sure I'm gonna be able to figure this one out. Having trouble going from the `decl` to figuring out if that declaration is inside a nested property. I'm not super familiar with the syntax tree used by stylelint, so that's probably just inexperience on my part.

**Comment by jeddy3** [comment]
@epmatsw Thanks for giving it a go.

It's actually my bad. I now think it should be a change to [`isStandardSyntaxDeclaration`](https://github.com/stylelint/stylelint/blob/master/src/utils/isStandardSyntaxDeclaration.js), rather than `isStandardSyntaxProperty`. In `isStandardSyntaxDeclaration` we have access to the `decl` _node_, whereas we just check the property _string_  in `isStandardSyntaxProperty`.

There's [a line](https://github.com/stylelint/stylelint/blob/master/src/utils/isStandardSyntaxDeclaration.js#L11-L12) in `isStandardSyntaxDeclaration` which shows how to grab the parent rule. 

And there's code for testing for the presence of a trailing semi in [this PR](https://github.com/stylelint/stylelint/pull/1885/files).

Did you want to give it another go?
